### ISSUE TITLE
League and Championship Promos

### ISSUE THREAD

**Issue by lfgcampos** [PokemonTCG/pokemon-tcg-data#92]
Any plans to include League and Championship Promos?

https://shop.tcgplayer.com/pokemon/league-and-championship-cards
https://www.trollandtoad.com/pokemon-championship-league-organized-play-promos/3427

**Comment by caresx** [comment]
I can contribute league card images from PTCGO.

**Comment by gfkauer** [comment]
I suggest including a "variants" property in all cards, maybe like this
`{
    "id": "sm8-172",
    "name": "Electropower",
    "imageUrl": "https://images.pokemontcg.io/sm8/172.png",
    "subtype": "Item",
    "supertype": "Trainer",
    "number": "172",
    "artist": "Eske Yoshinob",
    "rarity": "Uncommon",
    "series": "Sun & Moon",
    "set": "Lost Thunder",
    "setCode": "sm8",
    "text": [
      "During this turn, your Lightning Pokémon's attacks do 30 more damage to your opponent's Active Pokémon (before applying Weakness and Resistance)."
    ],
    "variants": [
      {
        "type": "reverse"
      },
      {
        "type": "League Cup Promo",
        "imageUrl": "https://images.pokemontcg.io/sm8/172c.png",
        "imageUrlHiRes": "https://images.pokemontcg.io/sm8/172c_hires.png"
      },
      {
        "type": "League Challange Promo",
        "number": "172a",
        "artist": "Ayoka Yoshida",
        "imageUrl": "https://images.pokemontcg.io/sm8/172a.png",
        "imageUrlHiRes": "https://images.pokemontcg.io/sm8/172a_hires.png"
      }
    ],
    "imageUrlHiRes": "https://images.pokemontcg.io/sm8/172_hires.png"
  }`

**Comment by mattcan** [comment]
Ugh, wish I read the issues more closely before starting down #119.

The approach I took for Yellow A alts was to add a whole new card, much of the data copied from the existing card. This was based on how the alt was done for [Delinquent](https://pokemontcg.io/cards/xy9-98b).

I think the variant approach is better as there is a vast reduction in duplication.

Any thoughts on this @adback03?

**Comment by ProfessorOaksLab** [comment]
@mattcan I just commented a long list of all missing cards on this issue: https://github.com/PokemonTCG/pokemon-tcg-data/issues/81 where I mention these types of promos.


To give my opinion on your question (even tho @adback03 should probably give the final answer).


I think these kinds of cards should be added as their own card to the API even tho the data is redundant. Why? Because they are release as different entites which means.

- They are valued differently
- One of them might be legal for play while the other is not
- They differ in print volume/rarity
- They might have different missprints
- They more often than not have different set symbol
- They more often than not have minor differences on the card
- A lot of sets, share cards with another set, it's how Pokemon TCG is built up, so by default the API would have a lot of "redundant" data. 

I've just now created a few other issues, on how we could show different "variants" of a card, that is actually different variants, and not different releases.

Adding a foil field: https://github.com/PokemonTCG/pokemon-tcg-data/issues/127
Adding a missprints field: https://github.com/PokemonTCG/pokemon-tcg-data/issues/129

I'm right now building an application that needs all this kinds of data. So I either need it added to the API or add it to my database. Doesn't really matter to me since it is the same amount of job, but I would much rather prefer that it was added to the API since we then can help out together to make sure the data is up-to-date.

In other words, I could help with these changes during the next couple of week, since I need to do them in one way or another. As long as we can agree that this is the way to go :)

**Comment by adback03** [comment]
Variants are definitely something I'd love to get into the API, but it is quite a huge undertaking. The data isn't super easy to come by, so it would need a large community effort to add it. Foil field and missprints are good ideas, but would need some help getting this info.
